contracts/shared/ComponentService.sol

```diff
@@ -585,30 +577,36 @@ contract ComponentService is
             IComponents.ProductInfo memory info
         )
     {
-        productNftId = instanceReader.getComponentInfo(componentNftId).productNftId;
+        productNftId = _getProductNftId(componentNftId);
         info = instanceReader.getProductInfo(productNftId);
     }
 
 
-    /// @dev based on the provided component address required type and role returns the component and related instance contract
-    /// the function reverts iff:
-    /// - the component has already been registered
+    /// @dev Based on the provided component address required type the component 
+    /// and related instance contract this function reverts iff:
+    /// - the sender is not registered
     /// - the component contract does not support IInstanceLinkedComponent
     /// - the component type does not match with the required type
-    /// - the initial component owner misses the required role (with the instance access manager)
+    /// - the component has already been registered
```

Comment:
`component.getInitialInfo()` is called twice, here and in registry service.
Potential source of inconsistency?

Reply:
added getRelease to IRegisterable, add checks in componentService